### PR TITLE
Add AWS EKS HyperPod receiver

### PR DESCRIPTION
**Pull request by petruanica**
#### Description
New receiver that collects node health status from SageMaker HyperPod clusters on EKS. Reads the sagemaker.amazonaws.com/node-health-status node label via the K8s API and emits 4 one-hot gauge metrics per node (hyperpod_node_health_status_schedulable, _unschedulable_pending_replacement, _unschedulable_pending_reboot, _unschedulable). Uses the scraperhelper pattern, default 60s poll interval.

#### Testing
24 tests passing — config validation, factory, scraper unit tests for all status values/edge cases, and 3 property-based tests with pgregory.net/rapid. gofmt and go vet clean.

#### Documentation
Added README.md with configuration, metrics table, attributes, prerequisites, and example RBAC.

### CODE REVIEW

File: receiver/awsekshyperpodreceiver/scraper.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+func (s *scraper) shutdown(_ context.Context) error {
+	s.logger.Info("Shutting down HyperPod health receiver")
+	if s.k8sClient != nil {
+		s.k8sClient.Shutdown()
+	}
+	return nil
+}
+
+func (s *scraper) scrape(_ context.Context) (pmetric.Metrics, error) {
+	nodeInfos := s.nodeClient.NodeInfos()
```
Comment (mitali-salvi) [inline receiver/awsekshyperpodreceiver/scraper.go]: seems like nodeInfos is unused and can be removed. its passed on to the processNode() function but unused

File: receiver/awsekshyperpodreceiver/scraper.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+		statusStr := status.String()
+		value := int64(0)
+		if statusStr == currentStatus {
+			value = 1
+		}
+
+		metricName := statusToMetricName[statusStr]
+		metric := sm.Metrics().AppendEmpty()
+		metric.SetName(metricName)
+		metric.SetDescription(fmt.Sprintf("HyperPod node health status: %s", statusStr))
```
Comment (mitali-salvi) [inline receiver/awsekshyperpodreceiver/scraper.go]: might wanna create an identical map of status to description
```
var statusToDescription = map[string]string{
    k8sutil.Schedulable.String():                    "HyperPod node health status: Schedulable",
    k8sutil.UnschedulablePendingReplacement.String(): "HyperPod node health status: UnschedulablePendingReplacement",
    k8sutil.UnschedulablePendingReboot.String():      "HyperPod node health status: UnschedulablePendingReboot",
    k8sutil.Unschedulable.String():                   "HyperPod node health status: Unschedulable",
}
```

and then directly calling 
```
metric.SetDescription(statusToDescription[statusStr])
```

File: receiver/awsekshyperpodreceiver/scraper_property_test.go [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+			if !found {
+				t.Fatalf("node %s: expected metric %s to be 1, but it wasn't", nodeName, expectedMetricName)
+			}
+		}
+	})
+}
+
+// --- Property 2: Attribute correctness ---
+// Feature: hyperpod-otel-contrib-receiver, Property 2: Attribute correctness
+// **Validates: Requirements 3.3, 5.2**
```
Comment (mitali-salvi) [inline receiver/awsekshyperpodreceiver/scraper_property_test.go]: what does this mean ?

Comment (petruanica) [reply receiver/awsekshyperpodreceiver/scraper_property_test.go]: Removed it

File: receiver/awsekshyperpodreceiver/scraper.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+func (s *scraper) scrape(_ context.Context) (pmetric.Metrics, error) {
+	nodeToLabelsMap := s.nodeClient.NodeToLabelsMap()
+
+	s.logger.Debug("Collected nodes",
+		zap.Int("nodesWithLabels", len(nodeToLabelsMap)),
+	)
+
+	metrics := pmetric.NewMetrics()
```
Comment (mitali-salvi) [inline receiver/awsekshyperpodreceiver/scraper.go:96]: is the expectation that this scraper will publish empty ResourceMetrics/ScopeMetrics when no nodes produce data ?

File: receiver/awsekshyperpodreceiver/scraper_property_test.go
Comment (mitali-salvi) [inline receiver/awsekshyperpodreceiver/scraper_property_test.go:1]: why is this a separate unit test file ?

File: receiver/awsekshyperpodreceiver/scraper_test.go [hunk 5]
```diff
@@ -0,0 +1,461 @@
...
+	"go.opentelemetry.io/collector/scraper/scraperhelper"
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/k8s/k8sclient"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/k8s/k8sutil"
+)
+
+// mockNodeClient implements k8sclient.NodeClient for testing.
+type mockNodeClient struct {
+	nodeInfos       map[string]*k8sclient.NodeInfo
```
Comment (mitali-salvi) [inline receiver/awsekshyperpodreceiver/scraper_test.go]: this seems unused

File: receiver/awsekshyperpodreceiver/scraper_property_test.go
Comment (petruanica) [reply receiver/awsekshyperpodreceiver/scraper_property_test.go:1]: Merged with `scraper_test.go`

File: receiver/awsekshyperpodreceiver/scraper_test.go (hunk 5 above)
Comment (petruanica) [reply receiver/awsekshyperpodreceiver/scraper_test.go]: Removed it

File: receiver/awsekshyperpodreceiver/scraper.go (hunk 4 above)
Comment (petruanica) [reply receiver/awsekshyperpodreceiver/scraper.go:96]: Changed it to return `pmetric.NewMetrics()` instead

File: receiver/awsekshyperpodreceiver/scraper.go [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+// statusToDescription maps each HyperPodConditionType string to its metric description.
+var statusToDescription = map[string]string{
+	k8sutil.Schedulable.String():                     "HyperPod node health status: Schedulable",
+	k8sutil.UnschedulablePendingReplacement.String(): "HyperPod node health status: UnschedulablePendingReplacement",
+	k8sutil.UnschedulablePendingReboot.String():      "HyperPod node health status: UnschedulablePendingReboot",
+	k8sutil.Unschedulable.String():                   "HyperPod node health status: Unschedulable",
+}
+
+// statusToMetricName maps each HyperPodConditionType string to its full metric name.
```
Comment (JayPolanco) [inline receiver/awsekshyperpodreceiver/scraper.go]: nit: from a code reading perspective, the global variables should be on top as they get called later

File: receiver/awsekshyperpodreceiver/scraper.go [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+}
+
+func newScraper(config *Config, settings receiver.Settings) *scraper {
+	return &scraper{
+		config: config,
+		logger: settings.Logger,
+	}
+}
+
+func (s *scraper) start(_ context.Context, _ component.Host) error {
```
Comment (mitali-salvi) [inline receiver/awsekshyperpodreceiver/scraper.go:61]: nit - missing code coverage

**Review by mitali-salvi — DISMISSED** [review]

File: receiver/awsekshyperpodreceiver/scraper.go [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+func (s *scraper) emitHealthMetrics(sm pmetric.ScopeMetrics, nodeName, instanceID, currentStatus string, timestamp pcommon.Timestamp) {
+	for _, status := range allStatuses {
+		statusStr := status.String()
+		value := int64(0)
+		if statusStr == currentStatus {
+			value = 1
+		}
+
+		metricName := statusToMetricName[statusStr]
+		metric := sm.Metrics().AppendEmpty()
```
Comment (JayPolanco) [inline receiver/awsekshyperpodreceiver/scraper.go]: If I am reading this correctly, this would essentially be creating multiple duplicate metrics within the ScopeMetrics. So it would look something like:

```
{
  "scopeMetrics": [{
    "metrics": [
      {
        "name": "hyperpod_node_health_status_schedulable",
        "gauge": {
          "dataPoints": [
            { "attributes": [{ "key": "node_name", "value": "node-1" }], "asInt": 1 }
          ]
        }
      },
      {
        "name": "hyperpod_node_health_status_unschedulable",
        "gauge": {
          "dataPoints": [
            { "attributes": [{ "key": "node_name", "value": "node-1" }], "asInt": 0 }
          ]
        }
      },
      {
        "name": "hyperpod_node_health_status_schedulable",
        "gauge": {
          "dataPoints": [
            { "attributes": [{ "key": "node_name", "value": "node-2" }], "asInt": 0 }
          ]
        }
      },
      {
        "name": "hyperpod_node_health_status_unschedulable",
        "gauge": {
          "dataPoints": [
            { "attributes": [{ "key": "node_name", "value": "node-2" }], "asInt": 1 }
          ]
        }
      }
    ]
  }]
}
```

In reality it should be something like this:
```
{
  "scopeMetrics": [{
    "metrics": [
      {
        "name": "hyperpod_node_health_status_schedulable",
        "gauge": {
          "dataPoints": [
            { "attributes": [{ "key": "node_name", "value": "node-1" }], "asInt": 1 },
            { "attributes": [{ "key": "node_name", "value": "node-2" }], "asInt": 0 }
          ]
        }
      },
      {
        "name": "hyperpod_node_health_status_unschedulable",
        "gauge": {
          "dataPoints": [
            { "attributes": [{ "key": "node_name", "value": "node-1" }], "asInt": 0 },
            { "attributes": [{ "key": "node_name", "value": "node-2" }], "asInt": 1 }
          ]
        }
      }
    ]
  }]
}
```

[Producers SHOULD prevent the presence of multiple Metric identities for a given name with the same Resource and Scope attributes](https://opentelemetry.io/docs/specs/otel/metrics/data-model/#opentelemetry-protocol-data-model-producer-recommendations)

Comment (petruanica) [reply receiver/awsekshyperpodreceiver/scraper.go]: Updated, thanks for catching this, let me know if it makes sense now

**Review by mitali-salvi — APPROVED** [review]

**Review by JayPolanco — APPROVED** [review]
